crates/bevy_reflect/src/serde/ser.rs

```diff
@@ -881,4 +895,43 @@ mod tests {
 
         assert_eq!(expected, bytes);
     }
+
+    #[test]
+    fn should_serialize_dynamic() {
```

Comment:
Could we rename this to be specific to `Option` since that's a special case in the serializer logic?
```suggestion
    fn should_serialize_dynamic_option() {
```